src/coreComponents/common/DataTypes.hpp

```diff
@@ -641,15 +643,31 @@ class rtTypes
     // [\\d]*  matches any number of numbers following the decimal
     // ([eE][-+]?[\\d]+|\\s*)  matches an optional scientific notation number
     // Note: the xsd regex implementation does not allow an empty branch, so use allow whitespace at the end
+    // TODO c++17: Move to static constexpr std::string_view
     string rr = "[+-]?[\\d]*([\\d]\\.?|\\.[\\d])[\\d]*([eE][-+]?[\\d]+|\\s*)";
 
-    // Regex to match a string that does not contain the characters  ,{}
-    string rs = "[^,\\{\\}]*";
+    // Regex to match a string that can't be empty and does not contain any whitespaces nor the characters ,{}
+    // TODO c++17: Move to static constexpr std::string_view
+    string rs = "[^,\\{\\}\\s]+\\s*";
```

Comment:
Do you need to take action with potential initial spaces (like ` toto`) or is it OK?

Reply:
This pattern will be permissive of spaces at the end of the string, but not at the end, which _should_ be fine, right?

```
"hello": match
"hello world": no match
"hello, world": no match
" hello": no match
"hello ": match
```